examples/employees/0_average-title-salary.prql

```diff
@@ -0,0 +1,23 @@
+# Task: rank the employee titles according to the average salary for each department.
+
+# My solution:
+# - for each employee, find their average salary,
+# - join employees with their departments and titles (duplicating employees for each of their titles and departments)
+# - group by department and title, aggregating average salary
+# - join with department to get department name
+
+from employees
+join side:left salaries [salaries.emp_no = employees.emp_no]
+aggregate by:[employees.emp_no] [
+  emp_salary: average salary
+]
+join side:left titles [titles.emp_no = table_0.emp_no]
+join dept_emp [dept_emp.emp_no = table_0.emp_no]
+aggregate by:[dept_emp.dept_no, titles.title] [
+  avg_salary: average emp_salary
+]
+join side:left departments [departments.dept_no = table_1.dept_no]
+select [dept_name, title, avg_salary]
+
+# Now writing about this I realize that I would have needed to select from employees table at all.
+# I could have only aggregated salaries table by emp_no. Oh well.
```

Comment:
Should we put this one in a separate PR and iterate? I think we should try and have the examples path full of really good quality queries — the others are great — and maybe this one isn't as efficient as those?